connectors/sources/mongo.py

```diff
@@ -196,14 +198,16 @@ def get_client(self):
             else:
                 client_params["tls"] = False
 
-            client = AsyncIOMotorClient(self.host, **client_params)
-
-            db = client[self.configuration["database"]]
-            self.collection = db[self.configuration["collection"]]
-
-            yield client
+            yield AsyncIOMotorClient(self.host, **client_params)
         finally:
-            self.remove_temp_file(certfile)
+            if os.path.exists(certfile):
+                try:
+                    os.remove(certfile)
+                except Exception as exception:
+                    self._logger.warning(
+                        f"Something went wrong while removing temporary certificate file. Exception: {exception}",
+                        exc_info=True,
+                    )
```

Comment:
Removed the helper method as it's very small and is pure.

Ideally I'd even delegate the file disposal to the function opening file instead of having our own custom code here instead.